api/v1/controllers/RSVPController.js

```diff
@@ -0,0 +1,152 @@
+
+var services = require('../services');
+var middleware = require('../middleware');
+var requests = require('../requests');
+var roles = require('../utils/roles');
+var mail = require('../utils/mail');
+
+var router = require('express').Router();
+
+function _isAuthenticated (req) {
+    return req.auth && (req.user !== undefined);
+}
+
+function _removeFromList(rsvpCurrent, rsvpNew) {
+    return rsvpCurrent.get('isAttending') && !rsvpNew.isAttending;
+}
+
+function _addToList(rsvpCurrent, rsvpNew) {
+    return !rsvpCurrent.get('isAttending') && rsvpNew.isAttending;
+}
+
+function createRSVP(req, res, next) {
+    if(!req.body.isAttending)
+        delete req.body.type;
+
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+           return services.RSVPService
+                .createRSVP(attendee, req.user, req.body);
+        })
+        .then(function(rsvp) {
+            if(rsvp.get('isAttending'))
+                services.MailService.addToList(req.user, mail.lists.attendees);
+            res.body = rsvp.toJSON();
+
+            return next();
+        })
+        .catch(function(error) {
+            return next(error);
+        });
+}
+
+function fetchRSVPByUser(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            return services.RSVPService
+                .findRSVPByAttendee(attendee);
+        })
+        .then(function (rsvp) {
+            res.body = rsvp.toJSON();
+            if(!res.body.type) {
+                delete res.body.type;
+            }
+
+            return next();
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function fetchRSVPByAttendeeId(req, res, next) {
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee);
+        })
+        .then(function(rsvp){
+            res.body = rsvp.toJSON();
+            if(!res.body.type) {
+                delete res.body.type;
+            }
+
+            return next();
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function updateRSVPByUser(req, res, next) {
+    if(!req.body.isAttending)
+        delete req.body.type;
+
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            return _updateRSVPByAttendee(req.user, attendee, req.body);
+        })
+        .then(function(rsvp){
+            res.body = rsvp.toJSON();
+
+            return next();
+        })
+        .catch(function (error) {
+            return next(error);
+        });
+}
+
+function updateRSVPById(req, res, next) {
+    if(!req.body.isAttending)
+        delete req.body.type;
+
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
```

Comment:
The `id` here should be the ID of the RSVP to update, not the RSVP of the attendee. Then you should have a method that finds an RSVP by ID.